src/main/java/com/redhat/devtools/lsp4ij/LanguageServersRegistry.java

```diff
@@ -384,25 +388,39 @@ private void removeAssociationsFor(LanguageServerDefinition definition) {
-                                                                                              @Nullable String configurationContent,
-                                                                                              @Nullable String initializationOptionsContent) {
-        updateServerDefinition(serverDefinition, name, commandLine, mappings, configurationContent, initializationOptionsContent, true);
-    }
-
-    @Nullable
-    public LanguageServerDefinitionListener.LanguageServerChangedEvent updateServerDefinition(@NotNull UserDefinedLanguageServerDefinition serverDefinition,
                                        @Nullable String name,
                                        @Nullable String commandLine,
+                                       @Nullable Map<String, String> userEnvironmentVariables,
+                                       boolean includeSystemEnvironmentVariables,
                                        @NotNull List<ServerMappingSettings> mappings,
                                        @Nullable String configurationContent,
-                                       @Nullable String initializationOptionsContent,
-                                       boolean notify) {
+                                       @Nullable String initializationOptionsContent) {
+        updateServerDefinition(serverDefinition,
+                name,
+                commandLine,
+                userEnvironmentVariables,
+                includeSystemEnvironmentVariables,
+                mappings,
+                configurationContent,
+                initializationOptionsContent,
+                true);
+    }
+
+    @Nullable
+    public LanguageServerDefinitionListener.LanguageServerChangedEvent updateServerDefinition(@NotNull UserDefinedLanguageServerDefinition serverDefinition,
+                                                                                              @Nullable String name,
+                                                                                              @Nullable String commandLine,
+                                                                                              @Nullable Map<String, String> userEnvironmentVariables,
+                                                                                              boolean includeSystemEnvironmentVariables,
+                                                                                              @NotNull List<ServerMappingSettings> mappings,
+                                                                                              @Nullable String configurationContent,
+                                                                                              @Nullable String initializationOptionsContent,
+                                                                                              boolean notify) {
```

Comment:
I think we're at a point where we're having too many parameters, it makes evolving code to complicated if we need to add one more parameter without breaking API or having to add countless overloads. I suggest we move to a single parameter object wrapping all those variables, e.g UpdateServerDefinitionQuery